file_catalog/schema/validation.py

```diff
@@ -36,26 +49,81 @@ def is_valid_sha512(hash_str: str) -> bool:
         """Check if `hash_str` is a valid SHA512 hash."""
         return re.match(r"[0-9a-f]{128}", str(hash_str), re.IGNORECASE) is not None
 
+    @staticmethod
+    def is_valid_location_list(locations: List[types.LocationEntry]) -> bool:
+        """Check if `locations` is a valid list of location-entries."""
+        if not isinstance(locations, list):
+            return False
+        if not locations:
+            return False
```

Comment:
In the second `if`, I'm looking to see if the list is empty.

Reply:
Technically the order doesn't matter, because a `None` will still work for `isinstance`.  But maybe it's even better to combine the two?

```python
if (not locations) or not isinstance(locations, list):
    return False
```

Reply:
Ahhh, I see. I was assuming it was like the input validation in `is_valid_location` below.
Considering `isinstance` does the work of null checking, then this is fine.

I prefer keeping it separate rather than combining it.
Same logic though, so it's just preference, not correctness. :shrug: